src/lammpsio/snapshot.py

```diff
@@ -176,19 +183,90 @@ def to_hoomd_gsd(self, type_map=None):
             frame.particles.mass = self.mass.copy()
         if self.has_molecule():
             frame.particles.body = self.molecule - 1
-
+        if self.has_bonds():
+            frame.bonds.N = self.bonds.N
+            frame.bonds.types = self.bonds.id
+            frame.bonds.group = self.bonds.members - 1
+            if self.bonds.has_typeid():
+                frame.bonds.typeid = numpy.zeros(self.bonds.N, dtype=int)
+                if bond_type_map is None:
+                    sorted_typeids = numpy.sort(numpy.unique(self.bonds.typeid))
+                    frame.bonds.types = [str(typeid) for typeid in sorted_typeids]
+                    for typeidx, typeid in enumerate(sorted_typeids):
+                        frame.bonds.typeid[self.bonds.typeid == typeid] = typeidx
+                else:
+                    frame.bonds.types = list(bond_type_map.values())
+                    reverse_bond_type_map = {
+                        typeid: typeidx
+                        for typeidx, typeid in enumerate(bond_type_map.keys())
+                    }
+                    for i, typeid in enumerate(self.bonds.typeid):
+                        frame.bonds.typeid[i] = reverse_bond_type_map[typeid]
```

Comment:
I understand why this is necessary, but should we make a method or function to do this rather than duplicating the code over and over?

Reply:
Yes, I agree it would be nice to have a helper function that does this.